watch.go

```diff
@@ -90,18 +92,67 @@ func liveEvents(lr Reloader, ch chan []string) {
 	}
 }
 
+func shouldInclude(file string, excludePatterns []string) bool {
+	for _, pattern := range excludePatterns {
+		match, err := filepath.Match(pattern, file)
+
+		if err != nil {
+			fmt.Println("Bad pattern: ", pattern)
+			continue
+		} else if match {
+			return false
+		}
+	}
+
+	return true
+}
+
 // WatchPaths watches a set of paths, and broadcasts changes through reloader.
-func WatchPaths(paths []string, reloader Reloader) error {
+func WatchPaths(paths, excludePatterns []string, reloader Reloader) error {
 	ch := make(chan []string, 1)
 	for _, path := range paths {
+		absPath, err := filepath.Abs(path)
+		if err != nil {
+			return err
+		}
+		if absPath[len(absPath)-1] != filepath.Separator {
+			absPath += string(filepath.Separator)
+		}
+
 		evtchan := make(chan notify.EventInfo, 1)
-		err := watch(path, evtchan)
+
+		err = watch(path, evtchan)
 		if err != nil {
 			return err
 		}
+
 		go func() {
 			for files := range batch(evtchan) {
-				ch <- files
+
+				if len(excludePatterns) > 0 {
+					empty := true
+					i := 0
+
+					for j, file := range files {
```

Comment:
Maybe extract this into a "filterPatterns(paths []string, excludes []string) []string" function, and unit test it?